### DISCUSSION TITLE
Complete callable alias as other command

### DISCUSSION THREAD

**Discussion by Qyriad** [xonsh/xonsh#4361] [Q&A] [answered]
Is there a way to use an existing completer for a given command for a callable alias? For example, I have an alias that wraps a Git command:
```xonsh
def _gac(name):
    git cherry-pick -n @(name)
    git reset
aliases["gac"] = _gac
```

Is there a way to make this callable alias tab complete its arguments as if the command were `git cherry-pick`? Or anything similar?

**Comment by gforsyth** [comment, marked as answer]
Hey @Qyriad -- we probably need a better way to do this, but I've done something similar with aliasing `kubectl` -> `k`:

```
bashcomp = __xonsh__.completers["bash"]

def _k_complete(prefix, line, start, end, ctx):
    if line.startswith("k "):
        return bashcomp(prefix, line.replace("k", "kubectl", 1), start, end, ctx)
    else:
        return None

__xonsh__.completers["k"] = _k_complete
__xonsh__.completers.move_to_end("k", last=False)
```

**Reply by Qyriad** [reply]
Could definitely use a better way to do this but this is pretty much exactly what I was looking for; thank you!
